test/MissionTest.t.sol

```diff
@@ -78,21 +98,229 @@ contract MissionTest is Test {
+        moonDAOTeamCreator.createMoonDAOTeam{value: 0.555 ether}("", "", "","name", "bio", "image", "twitter", "communications", "website", "view", "formId", new address[](0));
+        uint256 missionId = missionCreator.createMission(
+           0,
+           user1,
+           "",
+           0,
+           deadline,
+           1_000_000_000_000_000_000,
+           2_000_000_000_000_000_000,
+           true,
+           "TEST TOKEN",
+           "TEST",
+           "This is a test project"
+        );
+        uint256 projectId = missionCreator.missionIdToProjectId(missionId);
+
+        IJBTerminal terminal = jbDirectory.primaryTerminalOf(projectId, JBConstants.NATIVE_TOKEN);
+        uint256 balance = jbTerminalStore.balanceOf(address(terminal), projectId, JBConstants.NATIVE_TOKEN);
+        assertEq(balance, 0);
+
+        skip(2 days);
+        uint256 payAmount = 1_000_000_000_000_000_000;
+        vm.expectRevert();
+        terminal.pay{value: payAmount}(
+            projectId,
+            JBConstants.NATIVE_TOKEN,
+            0,
+            user1,
+            0,
+            "",
+            new bytes(0)
+        );
+    }
+
+    function testCreateTeamProjectBigPayment() public {
+        vm.startPrank(user1);
+        uint256 deadline = block.timestamp + 2 days;
+        moonDAOTeamCreator.createMoonDAOTeam{value: 0.555 ether}("", "", "","name", "bio", "image", "twitter", "communications", "website", "view", "formId", new address[](0));
+        uint256 missionId = missionCreator.createMission(
+           0,
+           user1,
+           "",
+           0,
+           deadline,
+           1_000_000_000_000_000_000,
+           2_000_000_000_000_000_000,
+           true,
+           "TEST TOKEN",
+           "TEST",
+           "This is a test project"
+        );
+        uint256 projectId = missionCreator.missionIdToProjectId(missionId);
+
+        IJBTerminal terminal = jbDirectory.primaryTerminalOf(projectId, JBConstants.NATIVE_TOKEN);
+        uint256 balance = jbTerminalStore.balanceOf(address(terminal), projectId, JBConstants.NATIVE_TOKEN);
+        assertEq(balance, 0);
+
+        uint256 payAmount = 2_000_000_000_000_000_000;
+        terminal.pay{value: payAmount}(
+            projectId,
+            JBConstants.NATIVE_TOKEN,
+            0,
+            user1,
+            0,
+            "",
+            new bytes(0)
+        );
+        uint256 balanceAfter1 = jbTerminalStore.balanceOf(address(terminal), projectId, JBConstants.NATIVE_TOKEN);
+        assertEq(balanceAfter1, payAmount);
+        uint256 tokensAfter1 = jbTokens.totalBalanceOf(user1, projectId);
+        //assertEq(tokensAfter1, 3_000 * 1e18);
+        // Due to rounding errors in calculating the weighted average, we end up with 3499999999999999999998 instead of the expected 3500000000000000000000
+        assertApproxEqAbs(tokensAfter1, 3_000 * 1e18, 2);
+    }
+
+    function testCreateTeamProjectHugePayment() public {
```

Comment:
Some other ideas for tests:

1.) currentFunding < min; payment > min but less than goal
2.) currentFunding < min; payment > goal
3.) currentFunding < goal; payment > goal
4.) currentFunding > goal; payment

Reply:
i think 1 is covered with the big payment test
2 is covered by huge payment test

i will add 3 and 4. and any other cases i can think of.